runner/internal/metrics/metrics.go

```diff
@@ -1,43 +1,67 @@
 package metrics
 
-import (
-	"time"
+import "reflect"
 
-	"github.com/benchttp/engine/runner/internal/recorder"
+type Value interface{}
+
+type Source string
+
+const (
+	ResponseTimeAvg     Source = "AVG"
+	ResponseTimeMin     Source = "MIN"
+	ResponseTimeMax     Source = "MAX"
+	RequestFailCount    Source = "FAILURE_COUNT"
+	RequestSuccessCount Source = "SUCCESS_COUNT"
+	RequestCount        Source = "TOTAL_COUNT"
 )
 
-// Aggregate is an aggregate of metrics resulting from
-// from recorded requests.
-type Aggregate struct {
-	Min, Max, Avg                          time.Duration
-	SuccessCount, FailureCount, TotalCount int
-	// Median, StdDev            time.Duration
-	// Deciles                   map[int]float64
-	// StatusCodeDistribution    map[string]int
-	// RequestEventsDistribution map[recorder.Event]int
-}
+type Type uint8
 
-// Compute computes and aggregates metrics from the given
-// requests records.
-func Compute(records []recorder.Record) (agg Aggregate) {
-	if len(records) == 0 {
-		return
+const (
+	TypeInt      Type = Type(reflect.Int)
+	TypeDuration Type = 12
+)
```

Comment:
Any particular reason for these enum values ?

Reply:
Nope, it was intended to be the largest `reflect.Kind` + 1. WIll make the replacement!